src/transformStateWithClones.js

```diff
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let stateCopy = { ...state };
+
+  for (const action of actions) {
+    const typeOfAction = action.type;
+
+    switch (typeOfAction) {
+      case 'addProperties':
+        stateCopy = {
+          ...stateCopy,
+          ...action.extraData,
+        };
+        break;
+
+      case 'removeProperties':
+        stateCopy = { ...stateCopy };
+
+        for (const key of action.keysToRemove) {
+          delete stateCopy[key];
+        }
+        break;
+
+      case 'clear':
+        stateCopy = {};
+
+        break;
+      default:
+        throw Error('Something went wrong');
+    }
+    result.push(stateCopy);
```

Comment:
I think you should spread `stateCopy` inside there
```suggestion
    result.push({ ...stateCopy });
```